SConstruct

```diff
@@ -381,6 +382,9 @@ if selected_platform in platform_list:
 
     suffix += env.extra_suffix
 
+    if env["arch"] == "":
+        env["arch"] = platform.machine()
```

Comment:
This should be done before `platform/***/detect.py -> configure(env)` is called here:
https://github.com/godotengine/godot/blob/2b0cb7e4c898b6094998af25ed1f47566f915f38/SConstruct#L320

Reply:
I've put it higher.
Note that this will affect the output binary filename, which was why I decided to put it there as to keep the current naming convention.

Reply:
This will indeed change binary filenames and break quite a few scripts.

Reply:
Would it be a good direction to adjust the naming convention to add the target architecture?

Maybe this could also replace the "bitness" and use or set it automatically:
__arm__, __armeb__, __mips__, __powerpc__ and __x86__ for __32-bit__, 
and 
__aarch64__, __arm64__, __mips64__, __powerpc64__ and __x86_64__ for __64-bit__?

Depending on the target architecture, maybe we could set the bitness from a list, if not specified?

(Ref: https://wiki.debian.org/Multiarch/Tuples - _multiarch name_ & _word size_)

Reply:
We do intend to refactor the current naming system to use proper triples (platform, arch, compiler) eventually, but that would be outside the scope of this PR (and not done before 3.1 as we're in feature freeze).

Reply:
I've opened #24030 about it.